File: src/main/java/com/matag/admin/MatagAdminWebConfiguration.java
package com.matag.admin;

import org.springframework.context.annotation.Configuration;
import org.springframework.web.servlet.config.annotation.ViewControllerRegistry;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;

@Configuration
public class MatagAdminWebConfiguration implements WebMvcConfigurer {
  @Override
  public void addViewControllers(ViewControllerRegistry registry) {
    registry.addViewController("/").setViewName("redirect:/ui");
    registry.addViewController("/ui").setViewName("forward:/admin.html");
    registry.addViewController("/ui/**").setViewName("forward:/admin.html");
  }
}

File: src/main/kotlin/com/matag/admin/MatagAdminWebConfiguration.kt
package com.matag.admin

import org.springframework.context.annotation.Configuration
import org.springframework.web.servlet.config.annotation.ViewControllerRegistry
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer

@Configuration
open class MatagAdminWebConfiguration : WebMvcConfigurer {
    override fun addViewControllers(registry: ViewControllerRegistry) {
        registry.addViewController("/").setViewName("redirect:/ui")
        registry.addViewController("/ui").setViewName("forward:/admin.html")
        registry.addViewController("/ui/**").setViewName("forward:/admin.html")
    }
}
